src/main/java/com/mraof/minestuck/block/TrajectoryBlock.java

```diff
@@ -0,0 +1,144 @@
+package com.mraof.minestuck.block;
+
+import net.minecraft.block.*;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.MobEntity;
+import net.minecraft.particles.ParticleTypes;
+import net.minecraft.pathfinding.PathNodeType;
+import net.minecraft.state.BooleanProperty;
+import net.minecraft.state.IntegerProperty;
+import net.minecraft.state.StateContainer;
+import net.minecraft.state.properties.BlockStateProperties;
+import net.minecraft.util.Direction;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.vector.Vector3d;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+import java.util.Random;
+
+/**
+ * When not crouching, players are subjected to movement by the block when walking on top of them. In addition, fall damage is cancelled when entities land on a upward facing trajectory block of more than power 6.
+ * While players can avoid the effects by crouching, while they are subject to them, they have limited movement of their own and cannot jump
+ */
+public class TrajectoryBlock extends MSDirectionalBlock
+{
+	public static final IntegerProperty POWER = BlockStateProperties.POWER;
+	public static final BooleanProperty POWERED = BlockStateProperties.POWERED; //used for texture purposes
+	
+	protected TrajectoryBlock(Properties properties)
+	{
+		super(properties);
+		this.registerDefaultState(stateDefinition.any().setValue(FACING, Direction.UP).setValue(POWER, 0).setValue(POWERED, false));
+	}
+	
+	@Override
+	public void fallOn(World worldIn, BlockPos pos, Entity entityIn, float fallDistance)
+	{
+		BlockState trajectoryState = worldIn.getBlockState(pos);
+		if(entityIn.isSuppressingBounce())
+		{
+			super.fallOn(worldIn, pos, entityIn, fallDistance);
+		} else if(trajectoryState.getValue(FACING) == Direction.UP && trajectoryState.getValue(POWER) > 6)
+		{
+			entityIn.causeFallDamage(fallDistance, 0.0F); //reduces damage if the trajectory block faces up and is powered a reasonable amount
+		}
+	}
+	
+	@Override
+	public void updateEntityAfterFallOn(IBlockReader worldIn, Entity entityIn)
+	{
+		if(entityIn.isSuppressingBounce())
+		{
+			super.updateEntityAfterFallOn(worldIn, entityIn);
+		} else
+		{
+			Vector3d entityMotion = entityIn.getDeltaMovement();
+			
+			if(entityMotion.y < 0.0D)
+			{
+				entityIn.setDeltaMovement(entityMotion.x, -entityMotion.y * 0.1D, entityMotion.z); //intended to reset player's falling momentum
+			}
+		}
+	}
+	
+	@Override
+	public void stepOn(World worldIn, BlockPos pos, Entity entityIn)
+	{
+		super.stepOn(worldIn, pos, entityIn);
+		BlockState blockState = worldIn.getBlockState(pos);
+		updatePower(worldIn, pos);
+		
+		int power = blockState.getValue(POWER);
+		double powerMod = power / 16D;
+		
+		if(power != 0 && !(blockState.getValue(FACING) == Direction.UP && power < 7))
```

Comment:
You're sometimes using 6 or 7 with comparisons against the power level. How about introducing some appropriately-named constant for it?